src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -728,6 +728,40 @@ object Desugar {
               }
             } yield in.Seqn(Vector(dPre, exprAss, clauseBody))(src)
 
+          case PGoStmt(exp) =>
+            def unexpectedExprError(exp: PExpression) = violation(s"unexpected expression $exp in go statement")
+
+            // nParams is the number of parameters in the function/method definition, and
+            // args is the actual list of arguments
+            def getArgs(nParams: Int, args: Vector[PExpression]): Writer[Vector[in.Expr]] = {
+              sequence(args map exprD(ctx)).map {
+                // go function chaining feature
+                case Vector(in.Tuple(targs)) if nParams > 1 => targs
+                case dargs => dargs
+              }
+            }
+
+            exp match {
+              case inv: PInvoke => info.resolve(inv) match {
+                case Some(p: ap.FunctionCall) => p.callee match {
+                  case ap.Function(_, st.Function(decl, _, _)) =>
+                    val func = functionD(decl)
+                    for { args <- getArgs(decl.args.length, p.args) } yield in.GoFunctionCall(func, args)(src)
+
+                  case ap.ReceivedMethod(recv, _, _, st.MethodImpl(decl, _, _)) =>
```

Comment:
it seems to me that the case for a method expression is missing.

Reply:
You are right

Reply:
(fixed in commit 89fcad2993e1125c984c9638da397a30afea850e)